src/sentry/static/sentry/app/components/modals/sentryAppPublishRequestModal.tsx

```diff
@@ -0,0 +1,133 @@
+import {Body, Header} from 'react-bootstrap/lib/Modal';
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
+import _ from 'lodash';
+import {Location} from 'history';
+
+import {SentryApp} from 'app/types';
+import {t} from 'app/locale';
+import Form from 'app/views/settings/components/forms/form';
+import FormModel from 'app/views/settings/components/forms/model';
+import JsonForm from 'app/views/settings/components/forms/jsonForm';
+
+class PublishRequestFormModel extends FormModel {
+  modal: SentryAppPublishRequestModal;
+
+  constructor(modal, ...args) {
+    super(...args);
+    this.modal = modal;
+  }
+
+  getTransformedData() {
+    const data = this.getData();
+    //map object to list of questions
+    const questionnaire = this.modal.formFields.map(field => {
```

Comment:
Shouldn't you have a list of fields in `this.fields`? Instead of having to bind `this.modal`

Reply:
@billyvg `this.fields` is a map with the field name as the key. We don't actually put the label in there. So unless I were to make the field name the full question, then I can't get the question from `this.fields`. Since the question is dynamic based on the scopes, I don't really see a better way.

Reply:
ahh.. what about if you use `this.fieldDescriptor`

Reply:
@billyvg I can use `this.fieldDescriptor` just fine if I use a normal text label. However, if I switch the label to a React element so I can get wrap the scopes with `<code>`, then I can't read the text from the label. The only thing I can think of is binding the modal then calling a function to get the scopes so I can get the plain text string of the label.

Reply:
I like #2 to avoid the circular dependency between the modal and the model.